rmw_fastrtps_cpp/src/type_support_common.cpp

```diff
@@ -88,20 +90,24 @@ bool TypeSupport::deserializeROSmessage(
   assert(ros_message);
   assert(impl);
 
-  // Deserialize encapsulation.
-  deser.read_encapsulation();
-
-  // If type is not empty, deserialize message
-  if (has_data_) {
-    auto callbacks = static_cast<const message_type_support_callbacks_t *>(impl);
-    return callbacks->cdr_deserialize(deser, ros_message);
+  try {
+    // Deserialize encapsulation.
+    deser.read_encapsulation();
+
+    // If type is not empty, deserialize message
+    if (has_data_) {
+      auto callbacks = static_cast<const message_type_support_callbacks_t *>(impl);
+      return callbacks->cdr_deserialize(deser, ros_message);
+    }
+
+    // Otherwise, consume dummy byte
+    uint8_t dump = 0;
+    deser >> dump;
+    (void)dump;
+  } catch (const eprosima::fastcdr::exception::Exception &) {
+    return false;
```

Comment:
what about using `RCUTILS_LOG_ERROR_NAMED` to notify the user what happened?

Reply:
If I may, I'm slight inclined towards `RMW_SET_ERROR_MSG` instead (see [here](https://github.com/ros2/rmw/blob/master/rmw/include/rmw/error_handling.h)).